### PR TITLE
tlaplus: 1.7.0 -> 1.7.1 base upon released jar file

### PR DESCRIPTION
**Pull request by florentc**
###### Motivation for this change

The current version is one year late compared to the latest stable release. In the previous tlaplus package, there was an issue with the tools when passing some command line arguments such as `-workers 4` for `tlc`. A java string method was not found and the program would not proceed correctly. 

###### Things done

This updates tlaplus to the most recent stable version and directly fetches the tlatools jar file from the official Github release page instead of building from the sources. This solves the command line arguments issue.

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -3908,6 +3908,12 @@
     githubId = 183879;
     name = "Florian Klink";
   };
+  florentc = {
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:3911]: Please cherry-pick the commit into the PR where you are adding yourself as a maintainer to a package. This makes sure all maintainers entries are actually used and we do not collect stale ones.

File: pkgs/applications/science/logic/tlaplus/default.nix [hunk 2]
```diff
@@ -1,41 +1,37 @@
...
+    url = "https://github.com/tlaplus/tlaplus/releases/download/v${version}/tla2tools.jar";
+    sha256 = "d532ba31aafe17afba1130f92410d9257454ff7393d1eb2fe032f0c07f352da5";
   };
 
   nativeBuildInputs = [ makeWrapper ];
-  buildInputs = [ adoptopenjdk-bin ant ];
+  buildInputs = [ adoptopenjdk-bin ];
 
-  buildPhase = "ant -f tlatools/org.lamport.tlatools/customBuild.xml compile dist";
+  unpackPhase = ":";
```
Comment (SuperSandro2000) [inline pkgs/applications/science/logic/tlaplus/default.nix:15]:
```suggestion
  dontUnpack = true;
```

File: maintainers/maintainer-list.nix (hunk 1 above)
Comment (florentc) [reply maintainers/maintainer-list.nix:3911]: Would you like me to split the commit into two commits (one to add myself as a maintainer and one to update the package) or open another pull request to add myself as maintainer?

Comment (SuperSandro2000) [reply maintainers/maintainer-list.nix:3911]: That was the wrong template... You can do that in this PR.

That is the right one:

----

Please add yourself as a maintainer by adding an alphabetically sorted entry to `maintainers/maintainer-list.nix` and commit this change into a separate commit with the message ``maintainers: add NAME`` into this PR.

Comment (florentc) [reply maintainers/maintainer-list.nix:3911]: This is done.

**Pull request merged**